be/src/column/schema.cpp

```diff
@@ -45,7 +45,11 @@ Schema::Schema(Schema* schema, const std::vector<ColumnId>& cids)
     auto ori_sort_idxes = schema->sort_key_idxes();
     std::unordered_set<ColumnId> scids(ori_sort_idxes.begin(), ori_sort_idxes.end());
     for (int i = 0; i < cids.size(); i++) {
-        DCHECK_LT(cids[i], schema->_fields.size());
+        if (cids[i] >= schema->_fields.size()) {
+            _fields.resize(_fields.size() - 1);
+            continue;
+        }
```

Comment:
In what cases would `cids[i] >= schema->_fields.size()`?

Reply:
If `cids[i] >= schema->_fields.size()` is true, `_fields` will be shrinked. Is it safe to access `_fields[j]` later, where `j` is greater than `i`?